File: libs/eave-stdlib-ts/src/core-api/operations/connect.ts
import { RequestArgsOrigin, makeRequest } from "../../requests.js";
import {
  ConnectInstallation,
  QueryConnectInstallationInput,
  RegisterConnectInstallationInput,
} from "../models/connect.js";
import { Team } from "../models/team.js";
import { CoreApiEndpointConfiguration } from "./shared.js";

export type RegisterConnectInstallationRequestBody = {
  connect_integration: RegisterConnectInstallationInput;
};
export type RegisterConnectInstallationResponseBody = {
  team?: Team;
  connect_integration: ConnectInstallation;
};

export class RegisterConnectInstallationOperation {
  static config = new CoreApiEndpointConfiguration({
    path: "/integrations/connect/register",
  });

  static async perform(
    args: RequestArgsOrigin & { input: RegisterConnectInstallationRequestBody },
  ): Promise<RegisterConnectInstallationResponseBody> {
    const resp = await makeRequest({
      config: this.config,
      ...args,
    });
    const responseData = <RegisterConnectInstallationResponseBody>(
      await resp.json()
    );
    return responseData;
  }
}

export type QueryConnectInstallationRequestBody = {
  connect_integration: QueryConnectInstallationInput;
};
export type QueryConnectInstallationResponseBody = {
  team?: Team;
  connect_integration: ConnectInstallation;
};

export class QueryConnectInstallationOperation {
  static config = new CoreApiEndpointConfiguration({
    path: "/integrations/connect/query",
  });

  static async perform(
    args: RequestArgsOrigin & { input: QueryConnectInstallationRequestBody },
  ): Promise<QueryConnectInstallationResponseBody> {
    const resp = await makeRequest({
      config: this.config,
      ...args,
    });
    const responseData = <QueryConnectInstallationResponseBody>(
      await resp.json()
    );
    return responseData;
  }
}

File: libs/eave-stdlib-py/src/eave/stdlib/core_api/operations/connect.py
from typing import Optional, Unpack
from eave.stdlib.api_types import BaseRequestBody, BaseResponseBody
from eave.stdlib.core_api.models import team
from eave.stdlib.core_api.models.connect import (
    ConnectInstallation,
    QueryConnectInstallationInput,
    RegisterConnectInstallationInput,
)

from . import CoreApiEndpoint, CoreApiEndpointConfiguration


from ... import requests


class RegisterConnectIntegrationRequest(CoreApiEndpoint):
    config = CoreApiEndpointConfiguration(
        path="/integrations/connect/register",
        auth_required=False,
        team_id_required=False,
        signature_required=True,
        origin_required=True,
    )

    class RequestBody(BaseRequestBody):
        connect_integration: RegisterConnectInstallationInput

    class ResponseBody(BaseResponseBody):
        team: Optional[team.Team]
        connect_integration: ConnectInstallation

    @classmethod
    async def perform(
        cls,
        input: RequestBody,
        **kwargs: Unpack[requests.CommonRequestArgs],
    ) -> ResponseBody:
        return await requests.make_request(
            config=cls.config,
            response_type=cls.ResponseBody,
            input=input,
            **kwargs,
        )


class QueryConnectIntegrationRequest(CoreApiEndpoint):
    config = CoreApiEndpointConfiguration(
        path="/integrations/connect/query",
        auth_required=False,
        team_id_required=False,
        signature_required=True,
        origin_required=True,
    )

    class RequestBody(BaseRequestBody):
        connect_integration: QueryConnectInstallationInput

    class ResponseBody(BaseResponseBody):
        team: Optional[team.Team]
        connect_integration: ConnectInstallation

    @classmethod
    async def perform(
        cls,
        input: RequestBody,
        **kwargs: Unpack[requests.CommonRequestArgs],
    ) -> ResponseBody:
        return await requests.make_request(
            config=cls.config,
            response_type=cls.ResponseBody,
            input=input,
            **kwargs,
        )
